buffer/src/main/java/io/netty/buffer/PoolArena.java

```diff
@@ -683,7 +694,10 @@ protected PoolChunk<byte[]> newUnpooledChunk(int capacity) {
 
         @Override
         protected void destroyChunk(PoolChunk<byte[]> chunk) {
-            // Rely on GC.
+            // Rely on GC. But keep one chunk for reuse.
+            if (!chunk.unpooled && lastDestroyedChunk.get() == null) {
+                lastDestroyedChunk.set(chunk); // The check-and-set does not need to be atomic.
```

Comment:
At this point can use `lazySet` - although I don't expect such to be an hot path...

Reply:
yeah I think it really doesn't matter here.